From 2494088958a8d7e40e35e5247a09f8597887f763 Mon Sep 17 00:00:00 2001
From: Jason Wessel <jason.wessel@windriver.com>
Date: Thu, 11 Oct 2018 10:44:10 +0200
Subject: [PATCH 05/21] daemon.c: Check exit code of chdir()

Stop the compile warning and fix the code to act on a chdir() failure.
If this one does fail something is very, very wrong.

Signed-off-by: Jason Wessel <jason.wessel@windriver.com>
---
 daemon.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/daemon.c b/daemon.c
index 046c540..9828718 100644
--- a/daemon.c
+++ b/daemon.c
@@ -981,7 +981,10 @@ int main(int argc, char **argv)
 	sigaction(SIGALRM, &act, NULL);
 
 	/* don't make directory we started in busy */
-	chdir("/");
+	if(chdir("/") < 0) {
+	    fprintf(stderr, "could not change working directory\n");
+	    daemon_exit(0);
+	}
 
 	/* detach from terminal */
 	if (opt_detach) {
-- 
2.42.0

